app/src/main/java/fr/free/nrw/commons/media/zoomControllers/loading/CircleProgressBarDrawable.java

```diff
@@ -0,0 +1,44 @@
+package fr.free.nrw.commons.media.zoomControllers.loading;
+
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.Rect;
+import android.graphics.RectF;
+
+import com.facebook.drawee.drawable.ProgressBarDrawable;
+
+public class CircleProgressBarDrawable extends ProgressBarDrawable {
```

Comment:
Why we are using a custom ProgressBar drawable, is this necessary?

Reply:
The Fresco API demands it and the drawable they use in their sample is maybe the one used here, seeing as it has the same error, [link](https://stackoverflow.com/a/39894537)

I'd prefer to not use it and just have the indeterminate

Reply:
@neslihanturan i just realised that SimpleDraweeView in the Contributions home page while uploading shows linear progress bar so i think it would be more appropriate to keep linear one for UI consistency, i will change it

@macgills i have added the progressbar because if the image size is heavy then it might take some time to load as @neslihanturan elucidated in the issue thread, i will replace circular progressbar with linear one while keeping the indeterminate spinner

Reply:
It seems really confusing as a user that my image is loading twice. I don't mind if the bar is linear or circular but there should only be one loading indication

Reply:
@macgills i completely agree with you but the problem is that the 
 - In-deterministic spinner would not account for the downloading of a big image file 
 - Keeping only progressBar will show a show black screen for 95% of the time as you correctly pointed out previously 

I think the best middle ground is to keep a linear progress bar which is much less intrusive and would fulfill the purpose of showing download progress,
What do you think about this :)

Reply:
By default show the indeterminate spinner, surely there is some callback `imageLoadComplete` where you can just hide it, the place that currently hides the progressdrawable.

I don't know fresco well, I'd use picasso or maybe coil nowadays

Reply:
Yeah just stop setting it Gone in `onIntermediateImageSet`

Reply:
This is how it looks with the changes:
**The blue bar at the bottom is the progress bar**
![GIF-200312_172652 1](https://user-images.githubusercontent.com/44129798/76519885-d11e2e00-6487-11ea-853a-0efccd40708a.gif)

Reply:
That works, just frustrating this scenario doesn't have better support in fresco.

Maybe if we had a good placeholderImage it would feel less weird to me?

Reply:
@macgills should i change anything else ?

Reply:
I'll leave final say with @neslihanturan 
I still think only using the ProgressBar makes for a better visual experience